### ISSUE TITLE
The github resolver should check if no patch was generated

### ISSUE THREAD

**Issue by neubig** [All-Hands-AI/openhands-resolver#181]
If no changes were made to the code, the github resolver currently fails with the following error, which also means that the github workflow returns an opaque error about the workflow failing.

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/openhands_resolver/send_pull_request.py", line 500, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/openhands_resolver/send_pull_request.py", line 489, in main
    process_single_issue(
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/openhands_resolver/send_pull_request.py", line 367, in process_single_issue
    make_commit(patched_repo_dir, resolver_output.issue, issue_type)
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/openhands_resolver/send_pull_request.py", line 140, in make_commit
    raise RuntimeError(f"Failed to commit changes: {result}")
RuntimeError: Failed to commit changes: CompletedProcess(args='git -C output/patches/issue_180 commit -m \'Fix issue #180: \'"\'"\'Bump version of openhands-ai to 0.11.0\'"\'"\'\'', returncode=1, stdout='HEAD detached at 65915b0\nnothing to commit, working tree clean\n', stderr='')
Copied repository to output/patches/issue_180
Patch applied successfully
Git user configured as openhands
```

To fix this we should:
1. Modify `send_pull_request.py` to check in the `make_commit()` function if the patch fails with a "nothing to commit, working tree clean" error, and print a more comprehensible error after "ERROR: ..." 
2. Modify the `.github/workflows/openhands-resolver.yml` file so that if the workflow calling `send_pull_request.py` fails, the error message "ERROR: ..." is read in, and passed back to the user through a github comment.

**Comment by malhotra5** [comment]
I would be willing to take a shot at this!

**Comment by neubig** [comment]
Fixed by #216 and #219
